tests/integrationtests/tst_perfparser.cpp

```diff
@@ -181,6 +181,90 @@ private slots:
         m_machineHostName = QSysInfo::machineHostName();
     }
 
+    void testFileErrorHandling()
+    {
+        PerfParser parser(this);
+        QSignalSpy parsingFailedSpy(&parser, &PerfParser::parsingFailed);
+        QString message;
+
+        const auto notThereFile = QStringLiteral("not_here");
+        parser.initParserArgs(notThereFile);
+
+        QCOMPARE(parsingFailedSpy.count(), 1);
+        message = qvariant_cast<QString>(parsingFailedSpy.takeFirst().at(0));
+        QVERIFY(message.contains(notThereFile));
+        QVERIFY(message.contains(QLatin1String("does not exist")));
+
+        const auto parentDirs = QStringLiteral("../..");
+        parser.initParserArgs(parentDirs);
+        // note: initializing parser args reset the attached spy counter
+        QCOMPARE(parsingFailedSpy.count(), 1);
+        message = qvariant_cast<QString>(parsingFailedSpy.takeFirst().at(0));
+        QVERIFY(message.contains(parentDirs));
+        QVERIFY(message.contains(QLatin1String("is not a file")));
+
+        const auto noRead = QStringLiteral("no_r_possible");
+        const auto shell = QStandardPaths::findExecutable(QStringLiteral("bash"));
```

Comment:
don't use bash and external processes for removing a file creating a new one and setting chmod on it - just use `QTemporaryFile` and the `QFile` API to set the permissions

Reply:
also, instead of testing two things one after the other in a bigger function, make this data driven too please - the only difference is whether the file exists or whether it's not readable - you can simply pass a bool to indicate these two states and handle it in the setup accordingly

Reply:
Did as requested, sadly the temp file cannot be created in the "data" function, so I've used a null string as filename to request that to be done in the test function.